src/main/java/com/mashape/unirest/http/HttpResponse.java

```diff
@@ -80,12 +81,17 @@ public HttpResponse(org.apache.http.HttpResponse response, Class<T> responseClas
 			try {
 				byte[] rawBody;
 				Long contentLength = responseEntity.getContentLength();
+
+				if (!ResponseUtils.isMultipartContentType(contentType)) {
```

Comment:
should this check instead be for chunked transfer-encoding? a response with `content-type:multipart/*` can still have a `content-length` header, while one with `transfer-encoding: chunked` would not.

Reply:
The idea was that we limit all requests, except for file downloads (for reasons that I forget, but I think it was because we still wanted to allow large downloads for bulk, etc.). File downloads over HTTP are basically 1:1 with `Content-Length: multipart`, so we exempt those from size limits.

`Transfer-Encoding: chunked` we definitely **should** check: it's orthogonal with what we want to put limits on. We just can't check it up-front, because we won't get a `Content-Length` in that case ("Transfer-Encoding: Chunked" is basically mutually exclusive with `Content-Length`). So, we will rely on the stream checking for that.